app/src/main/java/dev/arkbuilders/rate/data/db/PortfolioDao.kt

```diff
@@ -0,0 +1,72 @@
+package dev.arkbuilders.rate.data.db
+
+import androidx.room.Dao
+import androidx.room.Entity
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.PrimaryKey
+import androidx.room.Query
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import dev.arkbuilders.rate.domain.model.Asset
+import dev.arkbuilders.rate.domain.model.CurrencyCode
+import dev.arkbuilders.rate.domain.repo.PortfolioRepo
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Entity
+data class RoomAsset(
+    @PrimaryKey(autoGenerate = true)
+    val id: Long = 0,
+    val code: CurrencyCode,
+    val amount: Double,
+    val group: String?
+)
+
+@Dao
+interface PortfolioDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
```

Comment:
This could be replaced with `@Upsert` annotation
```suggestion
    @Upsert
```